cli/plugin/plugin.go

```diff
@@ -305,11 +306,93 @@ func (p *Plugin) PostBazel(bazelOutputPath string) error {
 	log.Debugf("Running post-bazel hook for %s/%s", p.config.Repo, p.config.Path)
 	cmd := exec.Command(scriptPath, bazelOutputPath)
 	// TODO: Prefix stderr output with "output from [plugin]" ?
+	cmd.Dir = path
 	cmd.Stderr = os.Stderr
 	cmd.Stdout = os.Stdout
 	return cmd.Run()
 }
 
+// Pipe streams the combined console output (stdout + stderr) from the previous
+// plugin in the pipeline to this plugin, returning the output of this plugin.
+// If there is no previous plugin in the pipeline then the original bazel output
+// is piped in.
+//
+// It returns the original reader if no output handler is configured for this
+// plugin.
+//
+// See cli/example_plugins/go_highlight/handle_bazel_output.sh for an example.
+func (p *Plugin) Pipe(r io.Reader) (io.Reader, error) {
+	path, err := p.Path()
+	if err != nil {
+		return nil, err
+	}
+	scriptPath := filepath.Join(path, "handle_bazel_output.sh")
```

Comment:
@siggisim I went with `handle_bazel_output.sh` instead of `on_output.sh` because I thought the latter made it sort of seem like the script would be called for each output line. But this new name might have the same confusion? :shrug: Open to suggestions